### PR TITLE
Configure ESLint and add basic CI pipelines

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Install and configure ESlint
- Add CI pipelines for main branch and PR
    - Use `pull_request_target` for security
    - Push Docker image
- Fix errors found in code (including warnings for Actions components; I kept Wizard with warnings as there are more changes incoming)
- Update all dependencies
- Prepare Docker image

## Testing

- Main branch build: https://github.com/pkosiec/dashboard/actions/runs/1616075670
- Sample PR: https://github.com/pkosiec/dashboard/pull/1
    - PR build: https://github.com/pkosiec/dashboard/actions/runs/1616077052

You can run built images with the following command:

```bash
docker run --rm -it -p 3000:80 ghcr.io/capactio/dashboard:7e48f12
# or 
docker run --rm -it -p 3000:80 ghcr.io/pkosiec/pr/dashboard:PR-1
```

> **NOTE:** Currently it is configured for local Capact installation (https://gateway.capact.local) with default password. Once we have login form implemented, then we can remove such temporary configuration.

### CODE REVIEW

File: .github/workflows/pr-build.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  build:
+    name: Build application
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    permissions:
+      contents: read
+    env:
+      DOCKER_TAG: PR-${{ github.event.pull_request.number }}
+      DOCKER_REPOSITORY: "ghcr.io/capactio/pr"
+      APP_NAME: "dashboard"
```
Comment (mszostok) [inline .github/workflows/pr-build.yaml:35]: You can export that to workflow envs as it's used between jobs 👍 

tested here: https://github.com/mszostok/dashboard/pull/1

File: scripts/push-image.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+DOCKER_REPOSITORY=${DOCKER_REPOSITORY:-"ghcr.io/capactio"}
+DOCKER_TAG=${DOCKER_TAG:-"latest"}
+
+CURRENT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+REPO_ROOT_DIR=$(cd "${CURRENT_DIR}/.." && pwd)
+APP_NAME="dashboard"
+
+pushd "${REPO_ROOT_DIR}"
+
+docker push ${DOCKER_REPOSITORY}/${APP_NAME}:${DOCKER_TAG}
```
Comment (mszostok) [inline scripts/push-image.sh:16]:
```suggestion
docker push "${DOCKER_REPOSITORY}"/${APP_NAME}:"${DOCKER_TAG}"
```

File: scripts/build-image.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+DOCKER_REPOSITORY=${DOCKER_REPOSITORY:-"ghcr.io/capactio"}
+DOCKER_TAG=${DOCKER_TAG:-"latest"}
+
+CURRENT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+REPO_ROOT_DIR=$(cd "${CURRENT_DIR}/.." && pwd)
+APP_NAME="dashboard"
+
+pushd "${REPO_ROOT_DIR}"
+
+docker build -t ${DOCKER_REPOSITORY}/${APP_NAME}:${DOCKER_TAG} .
```
Comment (mszostok) [inline scripts/build-image.sh:16]:
```suggestion
docker build -t "${DOCKER_REPOSITORY}"/${APP_NAME}:"${DOCKER_TAG}" .
```

File: README.md [hunk 4]
```diff
@@ -95,3 +95,31 @@ The build is minified and the filenames include the hashes.
...
+
+### Run built Docker image
+
+To run built Docker image, execute:
+
+```bash
+docker run --rm -it -p 3000:80 ghcr.io/capactio/dashboard:latest
+```
+
+The app listens on port `3000`.
```
Comment (mszostok) [inline README.md:125]:
```suggestion
To access Dashboard open [http://127.0.0.1:3000](http://127.0.0.1:3000).
```
or 
```suggestion
Now you can access Dashboard by opening [http://127.0.0.1:3000](http://127.0.0.1:3000).
```
or sth similar what gives us an option to easily open tab with dashboard :)

**Comment by pkosiec** [comment]
Improvements pushed: https://github.com/capactio/dashboard/pull/7/commits/ba135462f68db768cdbbc7de13ee094175e1c7cc
Tested:
PR: https://github.com/pkosiec/dashboard/actions/runs/1616077052
Main: https://github.com/pkosiec/dashboard/actions/runs/1618537851

**Review by mszostok — APPROVED** [review]

**Pull request merged**